File: Problems/No.404_Sum_of_Left_Leaves/sum_of_left_leaves.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int sumOfLeftLeaves(TreeNode* root) {
        
        // parameters
        std::queue<std::pair<TreeNode *, bool>> TNodes;
        int LL_sum = 0;

        // start computing
        TNodes.push({root, false});
        while (!TNodes.empty()) {

            auto [currNode, LoR] = TNodes.front();
            TNodes.pop();
            if ((currNode->left == nullptr) &&(currNode->right == nullptr))
                LL_sum += LoR ? currNode->val : 0;
            
            if (currNode->left) TNodes.push({currNode->left, true});
            if (currNode->right) TNodes.push({currNode->right, false});

        }

        return LL_sum;

    }
};

File: Problems/No.404_Sum_of_Left_Leaves/sum_of_left_leaves.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def sumOfLeftLeaves(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """

        # parameters
        TNodes = []
        LL_sum = 0

        # start computing
        TNodes.append([root, False])
        while TNodes:
            [currNode, LoR] = TNodes.pop(0)
            if ((currNode.left == None) and (currNode.right == None)):
                LL_sum += currNode.val if LoR else 0
            
            if currNode.left:
                TNodes.append([currNode.left, True])
            if currNode.right:
                TNodes.append([currNode.right, False])
        
        return LL_sum
